src/main/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -79,33 +98,184 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
 
         List<BibEntry> bibItems = new ArrayList<>();
 
-        // Obtain a factory object for creating SAX parsers
-        SAXParserFactory parserFactory = SAXParserFactory.newInstance();
-        // Configure the factory object to specify attributes of the parsers it
-        // creates
-        // parserFactory.setValidating(true);
-        parserFactory.setNamespaceAware(true);
-        // Now create a SAXParser object
-
         try {
-            SAXParser parser = parserFactory.newSAXParser(); //May throw exceptions
-            BibTeXMLHandler handler = new BibTeXMLHandler();
-            // Start the parser. It reads the file and calls methods of the handler.
-            parser.parse(new InputSource(reader), handler);
-            // When you're done, report the results stored by your handler object
-            bibItems.addAll(handler.getItems());
-
-        } catch (javax.xml.parsers.ParserConfigurationException e) {
+            JAXBContext context = JAXBContext.newInstance("net.sf.jabref.importer.fileformat.bibtexml");
+            XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+            XMLStreamReader xmlReader = xmlInputFactory.createXMLStreamReader(reader);
```

Comment:
Why can't directly the reader used as input for the unmarshaller?

The JavaDocs of the unmarshaller say so: https://jaxb.java.net/nonav/2.2.4/docs/api/javax/xml/bind/Unmarshaller.html

Reading that, the code would be much shorter and especially lines 106 to 109 would be obsolete